dataplane/src/pipeline/test_utils.rs

```diff
@@ -86,6 +87,7 @@ pub fn build_test_ipv4_packet(ttl: u8) -> Result<Packet<TestBuffer>, InvalidPack
     ipv4.set_source(UnicastIpv4Addr::new(Ipv4Addr::new(1, 2, 3, 4)).unwrap());
     ipv4.set_destination(Ipv4Addr::new(1, 2, 3, 4));
     ipv4.set_ttl(ttl);
+    ipv4.set_next_header_unchecked(NextHeader::UDP);
```

Comment:
Why is this function called `_unchecked`?

Reply:
There is a Ipv4::set_next_header() marked as unsafe. I believe the annotation is for the parsing case. I could use that one, but then nest it into an unsafe {} block.... but I see no point tbh. Ideally, the net crate would set this in the ip header when icmp/UDP/TCP etc. objects would be set... But that's not yet implemented, I believe. @daniel-noland ?

Reply:
I'm GTG once Daniel chimes in on this.

Reply:
If that helps, the ipv6 implementation is not unsafe:
```
    /// Set the next header [`IpNumber`]
    ///
    /// # Safety
    ///
    /// This method makes no attempt to ensure that the supplied [`next_header`] value is valid for
    /// the packet to which this header belongs (if any).
    ///
    /// [`next_header`]: NextHeader
    pub fn set_next_header(&mut self, next_header: NextHeader) -> &mut Self {
        self.0.next_header = next_header.0;
        self
    }
```
I can remove the unsafe annotation for the Ipv4 one and remove the _unchecked() method.
cc @mvachhar @daniel-noland

Reply:
@mvachhar I've done some corrections on the naming of NFs. I can break this PR into 3, but given the size of the PR I believe it's not worth it?